docs/src/pages/components/data-grid/editing/StartEditButtonGrid.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import { useGridApiRef, XGrid } from '@material-ui/x-grid';
+import {
+  randomCreatedDate,
+  randomTraderName,
+  randomUpdatedDate,
+} from '@material-ui/x-grid-data-generator';
+
+function EditToolbar({ buttonLabel, selectedCellParams, apiRef, setButtonLabel }) {
+  const handleButtonClick = React.useCallback(() => {
+    if (!selectedCellParams) {
+      return;
+    }
+    const { id, field, cellMode } = selectedCellParams;
+    if (cellMode === 'edit') {
+      const editedCellProps = apiRef.current.getEditCellPropsParams(id, field);
+      apiRef.current.commitCellChange(editedCellProps);
+      apiRef.current.setCellMode(id, field, 'view');
+      setButtonLabel('Edit');
+    } else {
+      apiRef.current.setCellMode(id, field, 'edit');
+      setButtonLabel('Save');
+    }
+    // Or you can use the editRowModel prop, but I find it easier with apiRef
+  }, [apiRef, selectedCellParams, setButtonLabel]);
+
+  return (
+    <div
+      style={{
+        justifyContent: 'center',
+        display: 'flex',
+        borderBottom: '1px solid rgba(224, 224, 224, 1)',
+      }}
+    >
+      <Button
+        onMouseDown={handleButtonClick}
```

Comment:
This is hiding a problem. Buttons should trigger on click.

```suggestion
        onClick={handleButtonClick}
```